### ISSUE TITLE
Only neural network model is working, error in threading when trying to predict with other models.

### ISSUE THREAD

**Issue by GiorgosSid** [kochlisGit/ProphitBet-Soccer-Bets-Predictor#94]
Whenever i try to predict fixtures with any model except the neural network it throws me this error:

![image](https://github.com/user-attachments/assets/6e5f0a07-b88a-4511-ae44-3ab895a48307)

Or when i am trying to evalute it shows me this error:

KeyError: 3
![image](https://github.com/user-attachments/assets/9310e415-d710-4c73-9ef4-73441f8d92ea)

I reinstalled everything with all the correct versions, with Cuda and Cudnn and i still get this error.

**Comment by kochlisGit** [comment]
That's interesting. Does it happen only with neural network? also, which league are you refering to?

**Comment by GiorgosSid** [comment]
> That's interesting. Does it happen only with neural network? also, which league are you refering to?
No, i am saying that the only machine learning model that works fine is the neural network, every other model it throws me an error. Also i am using the greece super league
